functions/src/modules/mewm/events/createEvent.ts

```diff
@@ -1,5 +1,59 @@
-import { ExpressFunction } from "../../../@types";
+import type { ExpressFunction } from "../../../@types";
+import { firebaseApp } from "../../../config/firebaseConfig";
+import {
+  expressErrorHandlerFactory,
+  requestBodyTypeValidator,
+} from "../../../helpers";
+import { uasPermissionSwitch } from "../../../systems/uas";
 
-const replaceMe: ExpressFunction = (req, res, next) => {};
+const createEvent: ExpressFunction = (req, res, next) => {
+  uasPermissionSwitch({
+    organizer: { accepted: validate },
+    sponsor: { accepted: validate }, // TO-DO: Need to filter for only sponsors where the benefit allows
+  })(req, res, next);
+};
 
-export default replaceMe;
+const validate: ExpressFunction = (req, res, next) => {
+  const validationRules: ValidatorObjectRules = {
+    type: "object",
+    rules: {
+      eventId: { rules: ["string"], required: true },
+      eventName: { rules: ["string"], required: true },
+      eventDescription: { rules: ["string"], required: true },
+      virtual: { rules: ["boolean"], required: true },
+      location: { rules: ["string"]},
+      joinLink: { rules: ["string"]},
+      joinLinkToPresenters: { rules: ["string"]},
+      joinLinkToAttendees: { rules: ["string"]},
```

Comment:
This is a number in the database, right?
```suggestion
      joinLinkToPresenters: { rules: ["number"]},
      joinLinkToAttendees: { rules: ["number"]},
```